### PR TITLE
Add z3_mk_atmost and z3_mk_atleast.

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: src/Z3/Base.hs [hunk 1]
```diff
@@ -2295,6 +2297,22 @@ getDatatypeSortConstructorAccessors c dtSort =
...
+
+mkAtLeast :: Context -> [AST] -> Int -> IO AST
+mkAtLeast _ctx [] _n = error "Z3.Base.mkAtLeast: empty list of expressions"
+mkAtLeast  ctx es  n =
+  marshal z3_mk_atleast ctx $ \f ->
+  marshalArrayLen es $ \esLen esArr ->
+  h2c n $ \n' ->
+  f esLen esArr n'
+
 -- TODO: Z3_func_decl_to_ast
```
Comment (maurobringolf) [inline src/Z3/Base.hs:2316]: @stevana Maybe you could insert the TODOs for `mk_pble`, `mk_pbge`, `mk_pbeq` too, so we have this section in order and up to date again?

https://z3prover.github.io/api/html/group__capi.html

File: test/Z3/Base/Spec.hs [hunk 2]
```diff
@@ -180,6 +180,71 @@ spec = around withContext $ do
...
+    specify "mkAtMost4" $ \ctx ->
+      (do
+        a <- Z3.mkFreshBoolVar ctx "A"
+        b <- Z3.mkFreshBoolVar ctx "B"
+        c <- Z3.mkFreshBoolVar ctx "C"
+        solver <- Z3.mkSolver ctx
+        Z3.solverAssertCnstr ctx solver =<< Z3.mkAtMost ctx [a, b, c] 4
+        (_r, Just model) <- Z3.solverCheckAndGetModel ctx solver
+        mapM (Z3.evalBool ctx model) [a, b, c])
+      `shouldReturn` [Just False, Just False, Just False]
```
Comment (maurobringolf) [inline test/Z3/Base/Spec.hs:214]: This is not necessarily guaranteed, or am I misreading something? Any assignment to the variables will satisfy the constraint, so this test could fail if z3 decides to choose another model.

File: src/Z3/Base.hs (hunk 1 above)
Comment (stevana) [reply src/Z3/Base.hs:2322]: Fixed!

File: test/Z3/Base/Spec.hs (hunk 2 above)
Comment (stevana) [reply test/Z3/Base/Spec.hs:214]: That's correct, should I just check for `Z3.Sat` and not bother with the model or implement `atLeast/Most :: [Maybe Bool] -> Bool` and assert that instead?

Comment (maurobringolf) [reply test/Z3/Base/Spec.hs:214]: Whatever you prefer, but in my opinion the second option is a bit more testing Z3 itself so personally I would just check sat in this case.

Comment (stevana) [reply test/Z3/Base/Spec.hs]: Ok, fixed!

**Review by IagoAbal — APPROVED** [review]
Looks good! Haven't run the tests but I trust they work. Thanks!

**Pull request merged**